### DISCUSSION TITLE
Changing last layer of Model

### DISCUSSION THREAD

**Discussion by VinaySingh561** [e3nn/e3nn#434] [Q&A] [answered]
Hi, 
I am using "https://docs.e3nn.org/en/stable/api/nn/models/gate_points_2101.html" with model_kwargs as following :
`model_kwargs = {
        "irreps_in": "2x 0e", #irreps_in 
        "irreps_hidden": [(mul, (l, p)) for l, mul in enumerate([125,40,25,15]) for p in [-1, 1]], #irreps_hidden
        "irreps_out": "1x0e", #irreps_out
        "irreps_node_attr": None, #irreps_node_attr
        "irreps_edge_attr": o3.Irreps.spherical_harmonics(3), #irreps_edge_attr
        "layers": 3,
        "max_radius": 3.5,
        "number_of_basis": 10,
        "radial_layers": 1,
        "radial_neurons": 128,
        "num_neighbors": 12.2298,
        "num_nodes": 24,
        "reduce_output": True, 
    }`...
After training this network i want to freeze all the weights and change the last layer in such a way that i get irrep out as ""12x0e + 5x1o + 4x2e + 2x3o + 1x4e" how should i do that?
Thank you for your help and consideration.

**Comment by mariogeiger** [comment, marked as answer]
Hello,

You can ask chatGPT the following question:

> How to freeze all the layers of my pytorch model and train only the last layer?

Unless you are asking how to change the irreps of the output? In that case I suggest to remove the last layer and retrain a new last layer from scratch.

**Comment by VinaySingh561** [comment]
Yes, I had to change the last layer from scratch, and as you suggested, ChatGPT worked its magic. It worked perfectly. Thank you!
